pycqed/measurement/detector_functions.py

```diff
@@ -965,17 +965,20 @@ def __init__(self, HS, threshold=1.75, trigger_separation=20e-6,
         self.last_frequency = 0.
         self.threshold = threshold
         self.last = 1.
-        self.trigger_separation = trigger_separation
-        self.demod_mode = demod_mode
+        HS.trigger_separation(trigger_separation)
```

Comment:
I did not remove these parameters from the detector function yet. I could do this but I think it will take quite a lot of testing from your side as well to make it work. Should I?